CRM/Speakcivi/Page/Speakcivi.php

```diff
@@ -180,6 +183,19 @@ public function share($param) {
 
   }
 
+  /**
+   * Create a representative mail activity
+   *
+   * @param $param
+   */
+  public function mail($param) {
+
+    $contact = $this->createContact($param);
+    $activity = $this->createActivity($param, $contact['id'], 'Email', 'Completed');
```

Comment:
@lolownia did you check that the email isn't sent twice (ie. not 100% sure it doesn't send the email)

Beside. improvement suggestion: you should add a "with" on that activity with the target (assuming it's a contact)